sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/PlanParserSuite.scala

```diff
@@ -852,6 +852,31 @@ class PlanParserSuite extends AnalysisTest {
         stop = 43))
   }
 
+  test("table valued function with named arguments") {
+    // All named arguments
+    assertEqual(
+      "select * from my_tvf(arg1 => 'value1', arg2 => true)",
+      UnresolvedTableValuedFunction("my_tvf",
+        NamedArgumentExpression("arg1", Literal("value1")) ::
+          NamedArgumentExpression("arg2", Literal(true)) :: Nil, Seq.empty).select(star()))
+
+    // Unnamed and named arguments
+    assertEqual(
+      "select * from my_tvf(2, arg1 => 'value1', arg2 => true)",
+      UnresolvedTableValuedFunction("my_tvf",
+        Literal(2) ::
+          NamedArgumentExpression("arg1", Literal("value1")) ::
+          NamedArgumentExpression("arg2", Literal(true)) :: Nil, Seq.empty).select(star()))
+
+    // Mixed arguments
+    assertEqual(
+      "select * from my_tvf(arg1 => 'value1', 2, arg2 => true)",
```

Comment:
Got it! Good point, added a test, it will throw: 
```
org.apache.spark.sql.AnalysisException: could not resolve `my_func` to a table-valued function; line 1 pos 14;
'Project [*]
+- 'UnresolvedTableValuedFunction [my_func], [jack, age => 18]
```

And the next step, we need to resolve this `UnresolvedTableValuedFunction` plan to actual function.